Naming.md

```diff
@@ -46,3 +53,54 @@ protocol FooEventHandling {
     // good
 }
 ```
+
+### Naming a delegate method
+
+The first parameter should always be the item that the delegate was called for.
+
+#### Example:
+
+```swift
+// really bad: there's no parameter for the item that the delegate was called for
+func didDeleteDraft(draft: Draft)
+
+// bad: the item that the delegate was called for is not the first parameter
+func didDeleteDraft(draft: Draft, draftManager: DraftManager)
+
+// good
+func draftManager(_ manager: DraftManager, didDeleteDraft draft: Draft)
+```
+
+### Naming an event handler method
+
+An event handler method should be name following this pattern:
+
+`handle<EventSource>[<Action>]`
+
+where `EventSource` is the UIControl or NSNotification which fired the event and `Action` is a past-tense verb described what happened.
+
+#### Example:
+
+```swift
+// Bad
+
+// An NSNotification handler (EventSource=KeyboardDidShowNotification, Action=NULL)
+didShowKeyboardNotification()
+
+// A UIButton handler (EventSource=ConfirmButton, Action=Tapped)
+confirmButtonWasTapped()
+
+// Good
+
+// An NSNotification handler (EventSource=KeyboardDidShowNotification, Action=NULL)
+handleKeyboardDidShowNotification()
+
+// A UIButton handler (EventSource=ConfirmButton, Action=Tapped)
+handleConfirmButtonTapped()
+```
+### Variables and Properties
+
+- Naming should be `camelCase`.
+- Variable names should always be descriptive with the exception of counter/index variables.
+- Include a type hint if the naming is ambiguous; e.g. `let confirm: Button` is bad, but `let confirmButton: Button` is good.
+- For booleans, prefixes such as `is` and `has` make it clear that a variable is a boolean and should be used.
```

Comment:
minor: could simplify the sentence to be more direct: "For booleans, use prefixes such as `is` and `has` make it clear that a variable is a boolean"